cinema/views.py

```diff
@@ -34,6 +36,18 @@
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        strings = ["actors", "genres", "title"]
+        for string in strings:
+            string = self.request.query_params.get(string)
+            if string:
+                string_ids = [int(str_id) for str_id in string.split(",")]
+                queryset = queryset.filter(id__in=string_ids).all()
+        if self.action == ("list", "retrieve"):
```

Comment:
The condition `if self.action == ("list", "retrieve"):` is incorrect. `self.action` is a string and should be compared to a single action at a time, such as `if self.action in ["list", "retrieve"]:`.